shivyc/tree/expr_nodes.py

```diff
@@ -798,6 +798,48 @@ class PostDecr(_IncrDecr):
     return_new = False
 
 
+class _ArithUnOp(_RExprNode):
+    """Base class for unary plus and minus."""
+
+    descrip = None
+    cmd = None
+
+    def __init__(self, expr):
+        """Initialize node."""
+        super().__init__()
+        self.expr = expr
+
+    def make_il(self, il_code, symbol_table, c):
+        """Make code for this node."""
+        expr = self.expr.make_il(il_code, symbol_table, c)
+        if expr.ctype.is_arith():
```

Comment:
Nit: to eliminate unnecessary code nesting, could you do reorder your statements like this instead:
```python
expr = ...
if not expr.ctype.is_arith():
  raise CompilerError(...)

if expr.ctype.size < 4:
  ...
```